wit/surface.wit

```diff
@@ -57,12 +58,187 @@ interface surface {
     }
 
     record key-event {
-        // should `code` and `key` be enums?
-        code: string,
-        key: string,
+        key: option<key>,
+        text: option<string>,
         alt-key: bool,
         ctrl-key: bool,
         meta-key: bool,
         shift-key: bool,
     }
+
+    enum key {
```

Comment:
Is this the list from https://w3c.github.io/uievents-code/#code-value-tables ? Perhaps document as such